source/docs/guide/src/nonlinear_bitvec.md

```diff
@@ -0,0 +1,287 @@
+# Integers: Nonlinear Arithmetic and Bit Manipulation
+
+Some properties about integers are very difficult (or expensive) to reason about fully automatically.
+As described below, to tackle these properties, Verus offers several dedicated proof strategies.
+
+# Nonlinear Arithmetic in Verus
+
+Nonlinear arithmetic involves equations that multiply, divide, or take the remainder of integer variables 
+(e.g., `x * (y * z) == (x * y) * z`).  As discussed earlier in this guide, determining the truth of such formulas
+is undecideable in general, meaning that general-purpose SMT solvers like Z3 can only make a best-effort attempt
+to solve them.  These attempts rely on heuristics that can be unpredictable.  Hence, by default, Verus 
+disables Z3's nonlinear arithmetic heuristics.  When you need to prove such properties, Verus offers the two dedicated
+proof strategies described below.  First, `integer_ring` feature can reliably prove a limited subset of nonlinear properties.  For properties
+outside that subset, Verus offers a way to invoke Z3's nonlinear heuristics in a way that will hopefully provide
+better reliability.
+
+## 1. Proving General Properties with Z3 
```

Comment:
It seems odd to start with the general case, rather than the specific case, especially since the last two sentences above do it in the other order.  Hence, I would swap the order of Z3 vs. Singular.